source/simulator/core.cc

```diff
@@ -217,6 +274,7 @@ namespace aspect
                                    boundary_indicator_lists[j].begin(),
                                    boundary_indicator_lists[j].end(),
                                    std::inserter(intersection, intersection.end()));
+
```

Comment:
I think the following AssertThrow now also checks the traction boundary conditions. So maybe change to: "[...] more than one type of velocity or traction boundary condition in the input file."